modules/apps/portal/portal-log4j-test/src/test/java/com/liferay/portal/log4j/Log4jTest.java

```diff
@@ -0,0 +1,464 @@
+
+import java.util.Date;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+import org.apache.log4j.SimpleLayout;
+import org.apache.log4j.WriterAppender;
+
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+/**
+ * @author Hai Yu
+ */
+public class Log4jTest {
+
+	@BeforeClass
+	public static void setUpClass() throws Exception {
+		PropsUtil.setProps(new PropsImpl());
+
+		_originalOutputStream = System.out;
+
+		_printStream = new TeePrintStream(_baos, _originalOutputStream);
+
+		System.setOut(_printStream);
+
+		Thread currentThread = Thread.currentThread();
+
+		ClassLoader classLoader = currentThread.getContextClassLoader();
+
+		URL url = classLoader.getResource(
+			"com/liferay/portal/log4j/dependencies/log4j.xml");
+
+		String urlContent = "";
+
+		try (InputStream inputStream = url.openStream()) {
+			byte[] bytes = _getBytes(inputStream);
+
+			urlContent = new String(bytes, StringPool.UTF8);
+		}
+
+		String dir = StringUtil.replace(
+			System.getProperty("user.dir"), '\\', '/');
+
+		urlContent = StringUtil.replace(urlContent, "@user_dir@", dir);
+
+		_tempFile = new File(System.getProperty("user.dir"), "/log4j.xml");
+
+		try {
+			_tempFile.createNewFile();
+
+			DataOutputStream outstream = new DataOutputStream(
+				new FileOutputStream(_tempFile));
+
+			outstream.write(urlContent.getBytes());
+
+			outstream.flush();
+
+			outstream.close();
+
+			URI uri = _tempFile.toURI();
+
+			Log4JUtil.configureLog4J(uri.toURL());
+
+			LogFactoryUtil.setLogFactory(new Log4jLogFactoryImpl());
+		}
+		catch (IOException ioException) {
+			throw ioException;
+		}
+	}
+
+	@AfterClass
+	public static void tearDownClass() {
+		_printStream.flush();
+
+		_printStream.close();
+
+		System.setOut(_originalOutputStream);
+
+		LogManager.shutdown();
+
+		if (_tempFile != null) {
+			_tempFile.delete();
+		}
+
+		File logDir = new File(
+			StringUtil.replace(System.getProperty("user.dir"), '\\', '/'),
+			"logs");
+
+		for (File file : logDir.listFiles()) {
+			file.delete();
+		}
+
+		logDir.delete();
+	}
+
+	@Test
+	public void testConsoleAppender() {
+		Log log = LogFactoryUtil.getLog(Log4jTest.class);
+
+		log.info("Test Message");
+
+		String[] logMessages = StringUtil.splitLines(_baos.toString());
+
+		String expectedOutput = logMessages[logMessages.length - 1];
+
+		_assertLog(expectedOutput, 147);
+	}
+
+	@Test
+	public void testGetOriginalLevel() {
+		String level = Log4JUtil.getOriginalLevel(_NAMES[5]);
+
+		Assert.assertEquals("The original level should be INFO", level, "INFO");
+
+		level = Log4JUtil.getOriginalLevel(_NAMES[6]);
+
+		Assert.assertEquals(
+			"The original level should be DEBUG", level, "DEBUG");
+	}
+
+	@Test
+	public void testLoggerEnabled() {
+		Log log = LogFactoryUtil.getLog(_NAMES[0]);
+
+		Assert.assertTrue("Logger should be all enabled", log.isTraceEnabled());
```

Comment:
Please see if it's possible to add a method like `_assertLevel(Log log, String level)` so that you can check exact levels. For example, for `Level.INFO` you should assert info is enabled but the one level above `info` is not.